### ISSUE TITLE
Custom renderer creation

### ISSUE THREAD

**Issue by iftu06** [handsontable/angular-handsontable#46]
### Description
How can i create a custom renderer? I can not find any register renderer method in angular-handson table .I cretaed a button but i can not trigger the action .

Here is the link
1.https://stackoverflow.com/questions/48052497/event-is-not-triggered-on-button-press-in-handsontable

**Comment by swistach** [comment]
Hi @iftu06,
Here is how you can use `renderer` option in settings object: https://stackblitz.com/edit/angular-handsontable-custom-renderer?file=app%2Fapp.component.ts

I found that a reference to the instance is missing when you try to get access to it using hook callback.

**Comment by iftu06** [comment]
thanks for demo,but in this demo , 'say hi from renderer'  button is not working.

**Comment by swistach** [comment]
You're right. It doesn't work right now. #22 is about of this feature.
I've added a missing description into the example from my last comment.

Moreover, I see here two different issues.
> How to create and use custom renderer?
You can find this in the example.

> How to use the angular component in custom renderers?
Currently, we don't allow this feature, you can find more information in #22

**Comment by AMBudnik** [comment]
Please track progress to https://github.com/handsontable/angular-handsontable/issues/46 for updates

**Comment by iftu06** [comment]
hi, if i send the schema from backend then it will not work,right? i am
keeping column scema in a json file in backend.i am sending it from backend
when request is made.
below json file is attached.



On Mon, Jan 15, 2018 at 1:59 PM, Aleksandra Budnik <notifications@github.com

**Comment by iftu06** [comment]
Hello, "How to use the angular component in custom renderers?" when this feature will be added.I really need this feature or can you suggest an alternate sollution??

**Comment by brynnlow** [comment]
Hello ,

can we hope to use ng-template to get access to Angular's own functionality ?

i mean something like this

<hot-column [data]="'id'" [title]="'Action'">
        <ng-template let-value="value" let-object="rowobject">
            <button (click)="makeMyAction(value)"></button>
            <img [src]="object.imagePath" alt="imagePath">
        </ng-template>
    </hot-column>

**Comment by swistach** [comment]
Hi @iftu06 and @brynnlow,
here is an example, based on angular tutorial: https://stackblitz.com/edit/angular-5ykm8c?file=src/app/heroes/heroes.component.ts

There is a solution, how to use component methods inside hooks.

**Comment by swistach** [comment]
We can close this issue if there is nothing more to add.

**Comment by ellipsistechnology** [comment]
Thanks very much @swistach for the example. However, note that the signature of beforeOnCellMouseDown seems to have changed in recent versions. I had trouble getting the example to work until I realised that the event parameter was actually a reference to the hotInstance. To get it working I changed as follows:
`beforeOnCellMouseDown: (hotInstance, event, coords, TD, blockCalculations) => {
        if (coords.col !== 2 || event.realTarget.nodeName !== "BUTTON") {
          return;
        }
        blockCalculations.cells = true;

        that.onSelect(hotInstance.getSourceDataAtRow(coords.row));
      }`
